presto-docs/src/main/sphinx/admin/properties.rst

```diff
@@ -151,6 +151,96 @@ Spilling Properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``boolean``
+    * **Default value:** ``true``
+
+    When ``spill_enabled`` is ``true``, this determines whether we will try spilling memory to disk for joins to
+    avoid exceeding memory limits for the query.
+
+    This config property can be overridden by the ``join_spill_enabled`` session property.
+
+``experimental.distinct-aggregation-spill-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``boolean``
+    * **Default value:** ``true``
+
+    When ``spill_enabled`` is ``true``, this determines whether we will try spilling memory to disk for distinct
+    aggregations to avoid exceeding memory limits for the query.
+
+    This config property can be overridden by the ``distinct_aggregation_spill_enabled`` session property.
+
+``experimental.order-by-aggregation-spill-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``boolean``
+    * **Default value:** ``true``
+
+    When ``spill_enabled`` is ``true``, this determines whether we will try spilling memory to disk for order by
+    aggregations to avoid exceeding memory limits for the query.
+
+    This config property can be overridden by the ``order_by_aggregation_spill_enabled`` session property.
+
+``experimental.spiller.task-spilling-strategy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    * **Type:** ``string``
+    * **Allowed values:** ``ORDER_BY_CREATE_TIME``, ``ORDER_BY_REVOCABLE_BYTES``, ``PER_TASK_MEMORY_THRESHOLD``
+    * **Default value:** ``ORDER_BY_CREATE_TIME``
+
+    Determines the strategy to use to choose when to revoke memory and from which tasks.
+
+    ``ORDER_BY_CREATE_TIME`` and ``ORDER_BY_REVOCABLE_BYTES`` will trigger spilling when the memory
+    pool is filled beyond the ``experimental.memory-revoking-threshold`` until the memory pool usage
+    is below ``experimental.memory-revoking-target``. ``ORDER_BY_CREATE_TIME`` will trigger
+    revocation from older tasks first, while ``ORDER_BY_REVOCABLE_BYTES`` will trigger revocation
+    from tasks that are using more revocable memory first.
+
+    ``PER_TASK_MEMORY_THRESHOLD`` will trigger spilling whenever the revocable memory used by a task
+    exceeds ``experimental.spiller.max-revocable-task-memory``.
+
+    .. WARNING::
+        The ``PER_TASK_MEMORY_THRESHOLD`` strategy does not trigger spilling when the memory pool is
+        full, which can prevent the out of memory query killer from kicking in.  This is particularly
+        risky if you are running without a reserved memory pool.
+
+``experimental.memory-revoking-threshold``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    * **Type:** ``double``
+    * **Default value:** ``0.9``
+
+    Trigger memory revocation when the memory pool is filled above this threshold.
+
+``experimental.memory-revoking-target``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    * **Type:** ``double``
+    * **Default value:** ``0.5``
```

Comment:
It's not clear to me what 0.5 refers to ie. the unit. is it a percentage / MB etc

Reply:
clarified that it's percentage